lib/msf/core/exploit/http/wordpress/login.rb

```diff
@@ -16,14 +16,39 @@ def wordpress_login(user, pass, timeout = 20)
     }, timeout)
     if res && res.redirect? && res.redirection && res.redirection.to_s == redirect
       cookies = res.get_cookies
-      # Check if a valid wordpress cookie is returned
-      return cookies if
-        # current Wordpress
-        cookies =~ /wordpress(?:_sec)?_logged_in_[^=]+=[^;]+;/i ||
-        # Wordpress 2.0
-        cookies =~ /wordpress(?:user|pass)_[^=]+=[^;]+;/i ||
-        # Wordpress 2.5
-        cookies =~ /wordpress_[a-z0-9]+=[^;]+;/i
+    end
+
+    if cookies && (
+      # current Wordpress (2.6+)
+      cookies =~ /wordpress_(?:sec|logged_in_)[^=]+=[^;]+;/i ||
+      # Wordpress 2.5
+      cookies =~ /wordpress_[a-z0-9]+=[^;]+;/i ||
+      # Wordpress 2.0
+      cookies =~ /wordpress(?:user|pass)_[^=]+=[^;]+;/i
+    )
+
+      service_data = {
+        address: rhost,
+        port: rport,
+        protocol: 'tcp',
+        service_name: 'http',
+        workspace_id: myworkspace_id,
+      }
+
+      cdata = {
+        module_fullname: self.fullname,
+        origin_type: :service,
+        username: user,
+        private_data: pass,
+        private_type: :password,
+      }.merge(service_data)
+
+      core = create_credential(cdata)
+      login_data = { core: core }.merge(service_data)
+
+      create_credential_login(login_data)
+
+      return cookies
```

Comment:
Lets revert file's changes if https://github.com/rapid7/metasploit-framework/pull/8327 is an acceptable alternative.